### PR TITLE
Enable contour unit conversion

### PR DESCRIPTION
**Pull request by rosteen**
This pulls out the contour-relevant code from #2702 and resolves a couple issues I ran into, as well as adding a test for the contour unit conversion. Requires https://github.com/glue-viz/glue-jupyter/pull/461 so we'll need a glue-jupyter release.

### CODE REVIEW

**Comment by rosteen** [comment]
Marked as ready for review, reminder that you need `glue-jupyter` `main` to test, and we merging is blocked by `glue-jupyter` release.

**Review by gibsongreen — APPROVED** [review]
Bear a glue release, these changes worked for me in testing, great job!

**Comment by pllim** [comment]
Are we then *stuck* until glue release happens? 😏 

p.s. Please rebase to remove change log conflict.

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (kecnry) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:288]: would it be hard to have this take `viewer` as an optional argument so that calls from the `AddToViewerMessage` only act on the single applicable viewer instead of looping over all viewers?

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (kecnry) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:302]: I'm guessing masks don't either.... could we instead check the units on the layer to make sure they're surface brightness?  Eventually we might need to generalize this method to also set display units for images in velocity, etc (from moment maps), but that isn't enabled in unit conversion yet, so is probably safe to ignore.

**Review by kecnry — COMMENTED** [review]
Seems to work well - thanks!

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (rosteen) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:288]: Hmm, but isn't unit conversion global, and would thus apply to contours in another viewer as well?

Comment (kecnry) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:288]: yes, but in theory when `AddDataToViewerMessage` is calling this, only _that_ viewer/layer combo needs to be updated, right?

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (kecnry) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:295]:
```suggestion
            viewers = self._app._viewer_store.values()
```

This will give dict values back, but since you're just iterating over it, it probably doesn't need to be casted to a list 🤷‍♂️

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (rosteen) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:308]: I added a second check below for surface brightness - we know DQ doesn't work, so might as well do the fast check first for that still?

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (kecnry) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:279]: what if `_find_and_convert_contour_units` just observed `sb_unit_selected` which already handles this logic?

Comment (kecnry) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:279]: we discussed that unit conversion probably needs some code moving soon anyways, so I'm fine deferring this until then

**Review by kecnry — APPROVED** [review]

**Comment by kecnry** [comment]
(looks like a test needs updating before merge, but assuming that is just the typo fix, approval stands)

**Comment by rosteen** [comment]
> (looks like a test needs updating before merge, but assuming that is just the typo fix, approval stands)

It's a real failure, I didn't account for subsets existing when units are converted. Should be a quick fix.

**Pull request merged**
